internal/services/compute/proximity_placement_group_resource.go

```diff
@@ -57,7 +56,7 @@ func resourceProximityPlacementGroup() *pluginsdk.Resource {
 				MinItems: 1,
 				Elem: &pluginsdk.Schema{
 					Type:         pluginsdk.TypeString,
-					ValidateFunc: validation.StringInSlice(virtualmachines.PossibleValuesForVirtualMachineSizeTypes(), false),
+					ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
We can't really do this here, whilst it may be lagging behind the swagger updates, not validating the values invites a lot of other problems.  Is there a reason there's a lag between the swagger updates and the service updates?

Reply:
This string slice only populates value for [HardwareProfile.vmSize](https://github.com/Azure/azure-rest-api-specs/blob/d66735f5dc80731ee5c7a31399c2a46f8743153a/specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2022-03-01/computeRPCommon.json#L437-L439) but not the full list of VM Sizes, thus some of the possible values are not contained there.

Reply:
does that not indicate that the swagger is incomplete and needs these missing values added?

Reply:
It seems like the service is [moving away from the static enum list](https://github.com/Azure/azure-rest-api-specs/blob/d66735f5dc80731ee5c7a31399c2a46f8743153a/specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2022-03-01/computeRPCommon.json#L441), so the list is incomplete any more. It is said in the swagger that "*The enum data type is currently deprecated and will be removed by December 23rd 2023.*".

Reply:
ahh then that would explain why it is not maintained. there isn't much we can do about it then. thanks @myc2h6o